src/ex1/script.js

```diff
@@ -0,0 +1,68 @@
+// trigger fork
+let count = 0;
+
+function addTodo(){
+    const todo = document.getElementById("newTodo").value;
+    if(todo.length === 0){
+        alert("Please enter a todo task")
+    }
+    else {
+        const todoList = document.getElementById("todoList");
+        const li = document.createElement("li");
+        li.id = 'todo' + count;
+        li.innerHTML = `<input type='checkbox' name='selectTodo' class= pl value="${todo}">`;
+        li.innerHTML += todo;
+        todoList.appendChild(li);
+        document.getElementById("newTodo").value = "";
+        count++;
+        console.log("is this checked" + li.checked);
+    }
+}
+document.getElementById("addTodo").addEventListener("click", addTodo);
+
+
+let history = [];
+function deleteTodo(){
+    let todos = document.querySelectorAll('.pl');
+    const todoList = document.getElementById("todoList");
+    todos.forEach(element => {
+        console.log("checked is " + element.checked);
+        if(element.checked === true){
+            todoList.removeChild(document.getElementById(element.parentElement.id));
+            history.push(element.parentElement);
+            element.checked = false;
+        }
+    })
+}
+
+document.getElementById("delete").addEventListener("click",deleteTodo);
+
+
+// Event listeners for keypress
+window.addEventListener("keyup",e =>{
+    const todoList = document.getElementById("todoList");
+    //enable ctrl+z for adding delete items
+    if(e.ctrlKey && e.key === 'z'){
+        if(history.length === 0){
+            alert("History is currently empty")
+        }
+        else{
+            todoList.appendChild(history.pop());
+        }
+    }
+    //enable submitting todo via the enter key
+    if(e.key==='Enter'){
```

Comment:
Do it only if I'm focused on the input element.. (Even better is to add this specific event listener to the input itself)
One more benefit of creating an event listener for the input itself is that you can do e.stopPropagation() and then if you're in the input, pressing ctrl+z won't propagate to the window and then won't trigger the history event